src/main/java/co/cask/PluginJSON.java

```diff
@@ -216,56 +213,69 @@ public void execute()  throws MojoExecutionException {
+  private void addIcon(JSONObject finalObject, Map<String, File> files, File nameFile) throws IOException {
+    JSONObject innerObject = new JSONObject();
     String name = FileUtils.removeExtension(nameFile.getName());
-    if (fileNames.contains(name)) {
-      object.put("type", "inline");
+    if (files.containsKey(name)) {
+      innerObject.put("type", "inline");
       Map<String, String> arguments = new TreeMap<String, String>();
-      File file = files[fileNames.indexOf(name)];
-      try {
-        arguments.put("data", getDataURISchemaAsString(file, "data", "image/" + FileUtils.getExtension(
-          file.getName()), true));
-      } catch (Exception e) {
-        throw new MojoExecutionException(
-          String.format("Issue converting icon file '%s' to Base64 format. %s", file.getName(), e.getMessage())
-        );
-      }
-      object.put("arguments", arguments);
+      File file = files.get(name);
+      String mediaType = URLConnection.guessContentTypeFromName(file.getName());
+      arguments.put("data", getDataURISchemaAsString(file, mediaType));
+      innerObject.put("arguments", arguments);
+      finalObject.put("icon", innerObject);
     }
-
-    return object;
   }
 
   /**
    * Returns the Data URI Scheme of File
    */
-  private String getDataURISchemaAsString(File file, String scheme, String mediaType, boolean base64Extension)
-    throws URISyntaxException, IOException {
-    String path = scheme + ":" + mediaType;
-    path = base64Extension ? path.concat(";base64,") : path;
+  private String getDataURISchemaAsString(File file, String mediaType)
+    throws IOException {
+    String uri = "data:" + mediaType + ";base64,";
+
+    FileInputStream inStream = null;
+    ByteArrayOutputStream outStream = null;
+    Base64OutputStream outStreamBase64 = null;
 
-    FileInputStream dataInFile = new FileInputStream(file);
-    byte[] dataInFileBytes = new byte[(int) file.length()];
-    dataInFile.read(dataInFileBytes);
-    path = path.concat(new String(Base64.encodeBase64(dataInFileBytes)));
+    try {
+      inStream = new FileInputStream(file);
+      outStream = new ByteArrayOutputStream();
+      outStreamBase64 = new Base64OutputStream(outStream);
+
+      byte[] inputData = IOUtils.toByteArray(inStream);
+      outStreamBase64.write(inputData);
+      outStreamBase64.flush();
+
+      String result = outStream.toString();
+      uri = uri.concat(result);
+    } finally {
+      outStream.flush();
+      outStreamBase64.close();
+      outStream.close();
+      inStream.close();
+    }
 
-    return path;
+    return uri;
   }
 
   /**
-   * Remove extensions from files in directory and return the names as a list
+   * Remove extensions from files in directory and return a map of names to files.
    */
-  private List<String> getFileNamesFromDirectory(File[] listFiles) {
-    List<String> names = new ArrayList<String>();
+  private Map<String, File> getFileNamesFromDirectory(File[] listFiles) {
```

Comment:
nit: rename to getFileNameMap or something like that. you're not getting file names from a directory, since you're passing in all the files.